crates/compiler/can/src/suffixed.rs

```diff
@@ -23,14 +23,23 @@ fn next_unique_suffixed_ident() -> String {
 
         // # is used as prefix because it's impossible for code authors to define names like this.
         // This makes it easy to see this identifier was created by the compiler.
-        format!("#!a{}", count)
+        format!("#!{}", count)
     })
 }
 
 #[derive(Debug)]
+///
 pub enum EUnwrapped<'a> {
+    /// First suffixed expression in a definition. Emitted if an expression has def pattern
+    /// e.g. x = first! (second! 42)
+    /// The first unwrap will produce
+    /// `UnwrappedDefExpr<first (second! 42)>`
     UnwrappedDefExpr(&'a Loc<Expr<'a>>),
 
+    /// Suffixed sub expression
+    /// e.g. x = first! (second! 42)
+    /// In this example, the second unwrap (after unwrapping the top level `first!`) will produce
+    /// `UnwrappedSubExpr<{ sub_arg: second 42, sub_pat: #!a0, sub_new: #!a0 }>`
```

Comment:
Minor nit. We change the unique prefix above, so these comments are now not quite right.

Otherwise these comments are a helpful addition.

Reply:
Nice. I updated the other comments, seems to miss this one

Reply:
fixed in https://github.com/roc-lang/roc/pull/6868/commits/3b50a8322a76c3e8f59d3cba555e2bc2754e260d